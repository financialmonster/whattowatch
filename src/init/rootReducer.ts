import { connectRouter } from 'connected-react-router/immutable';
import { combineReducers } from 'redux';
import { createBrowserHistory } from 'history';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import immutableTransform from 'redux-persist-transform-immutable';

import authReducer from 'domains/auth/authReducer';
import promoReducer from 'domains/promo/promoReducer';
import filmsReducer from 'domains/films/filmsReducer';
import reviewsReducer from 'domains/reviews/reviewsReducer';
import favoritesReducer from 'domains/favorites/favoritesReducer';

const persistConfig = {
    transforms: [immutableTransform()],
    key: `root`,
    storage
}

export const history = createBrowserHistory();
const routerReducer = connectRouter(history);

const rootReducer = combineReducers({
    router: routerReducer,
    auth: authReducer,
    promo: promoReducer,
    films: filmsReducer,
    reviews: reviewsReducer,
    favorites: favoritesReducer
});

export const persistedRootReducer = persistReducer(persistConfig, rootReducer);
