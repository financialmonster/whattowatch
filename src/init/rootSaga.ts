import { all, call } from 'redux-saga/effects';
import { SagaIterator } from '@redux-saga/core';

import { watchPromo } from 'domains/promo/saga/promoWatchers';
import { watchFilms } from 'domains/films/saga/filmsWatchers';
import { watchAuth } from 'domains/auth/saga/authWatchers';
import { watchReviews } from 'domains/reviews/saga/reviewsWatchers';

export function* rootSaga(): SagaIterator {
    yield all([call( watchPromo ), call( watchFilms ), call( watchAuth ), call( watchReviews )]);
}
