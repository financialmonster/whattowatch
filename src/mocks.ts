import { Map, fromJS, List } from 'immutable';

export const createStoreMock = (mode: string = `common`) => {
    return {
        promo: Map({
            promo: Map(),
            isPromoFetching: true,
            promoError: null
        }),
        films: Map({
            films: fromJS([{genre: `Thriller`, id: 1}, {genre: `Comedy`, id: 2}]),
            isFilmsFetching: mode === `detailed`,
            filmsError: null,
            filter: `All genres`
        }),
        auth: Map({
            user: null,
            authError: null,
            isAuthFetching: false,
            authStatusError: null,
            isAuthStatusFetching: false
        }),
        reviews: Map({
            reviews: List(),
            isReviewsFetching: true,
            reviewsError: null
        })
    }
}
