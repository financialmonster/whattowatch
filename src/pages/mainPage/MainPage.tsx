import React from 'react';

import Header from 'components/header/Header';
import PromoContainer from 'components/promoContainer/PromoContainer';
import { FilmsListContainer } from 'components/filmsListContainer/FilmsListContainer';

const MainPage = () => {
    return (
        <div className="main">
            <section className="movie-card">
                <div className="movie-card__bg">
                    <img src="img/bg-the-grand-budapest-hotel.jpg" alt="The Grand Budapest Hotel" />
                </div>
                <h1 className="visually-hidden">WTW</h1>
                <Header />
                <PromoContainer />
                <FilmsListContainer />
            </section>
        </div>
    );
};

export default MainPage;
