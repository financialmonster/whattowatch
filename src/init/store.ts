import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { routerMiddleware as createRouterMiddleware } from 'connected-react-router';

import { rootReducer, history } from './rootReducer';

const routerMiddleware = createRouterMiddleware(history);
const middlewares = [ routerMiddleware ];

const logger = createLogger({
    duration: true,
    collapsed: true,
    colors: {
        title: (action) => (action.error) ? 'firebrick' : 'deepskyblue',
        prevState: () => '#1C5FAF',
        action:    () => '#149945',
        nextState: () => '#A47104',
        error:     () => '#ff0005',
    }
});

if(process.env.NODE_ENV !== 'production') {
    middlewares.push(logger);
}

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)));

export default store;
