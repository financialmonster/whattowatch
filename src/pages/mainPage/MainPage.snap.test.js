
import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { routerMiddleware as createRouterMiddleware } from 'connected-react-router';

import { MainPage } from './MainPage';
import { createStoreMock } from 'mocks';
import { history } from 'init/rootReducer'; 

it(`MainPage: correctly rendered`, () => {
    const routerMiddleware = createRouterMiddleware(history);
    const mockStore = configureStore([ routerMiddleware ]);
    const store = mockStore(createStoreMock());

    const tree = renderer.create(
        <Provider store={store}>
            <Router history={history}>
                <MainPage />
            </Router>
        </Provider>        
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
