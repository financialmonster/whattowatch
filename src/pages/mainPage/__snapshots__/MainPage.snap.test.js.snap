// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage: correctly rendered 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          onClick={[Function]}
          style={Object {}}
          title=""
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__link"
          onClick={[Function]}
          title="To the login page"
        >
          Sign in
        </a>
      </div>
    </header>
    <div
      className="sc-AxjAm fcmpGc"
    >
      <div
        className="sc-AxirZ fLbCRQ"
        rotate="rotate(0deg)"
      />
      <div
        className="sc-AxirZ iRnakP"
        rotate="rotate(30deg)"
      />
      <div
        className="sc-AxirZ jJDLSA"
        rotate="rotate(60deg)"
      />
      <div
        className="sc-AxirZ cfISCN"
        rotate="rotate(90deg)"
      />
      <div
        className="sc-AxirZ jYaMud"
        rotate="rotate(120deg)"
      />
      <div
        className="sc-AxirZ kCauMN"
        rotate="rotate(150deg)"
      />
      <div
        className="sc-AxirZ bbuhJx"
        rotate="rotate(180deg)"
      />
      <div
        className="sc-AxirZ RfLVX"
        rotate="rotate(210deg)"
      />
      <div
        className="sc-AxirZ iDvjl"
        rotate="rotate(240deg)"
      />
      <div
        className="sc-AxirZ fchBdo"
        rotate="rotate(270deg)"
      />
      <div
        className="sc-AxirZ fGeryM"
        rotate="rotate(300deg)"
      />
      <div
        className="sc-AxirZ eVSKan"
        rotate="rotate(330deg)"
      />
    </div>
  </section>,
  <div
    className="page-content"
    style={
      Object {
        "backgroundImage": "linear-gradient(-180deg, undefined 0%, undefined 100%)",
      }
    }
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
            title=""
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
            style={Object {}}
            title="Comedy"
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
            style={Object {}}
            title="Thriller"
          >
            Thriller
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <img
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
            />
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <img
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
            />
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <img
              height="175"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
            />
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      />
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          onClick={[Function]}
          style={Object {}}
          title=""
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
